### DISCUSSION TITLE
iPadOS trackpad doesn’t scroll

### DISCUSSION THREAD

**Discussion by huw** [community/community#24966] [Codespaces]
<p>I’m on my iPad, with the Magic Keyboard, and was hoping to use Codespaces for dev work. But when I try to scroll (with the trackpad) inside an editor window, or any other panel, it won’t move. Scrolling with my fingers works fine.</p>

**Comment by matthewisabel** [comment]
<p>Thanks for reporting this, we’re tracking it and we’ll circle back when it’s fixed.</p>

**Comment by alefragnani** [comment]
<p>I’m having the same “no scrolling support” using regular Mouse Wheel (connected via Bluetooth to my iPad). Is this issue tracking this too, or should I open a new one?</p>

**Comment by lostintangent** [comment]
<p><a class="mention" href="/alefragnani">@alefragnani</a> Are you referring to a scrolling issue on a non-iPadOS device? It’s likely that the scrolling issue is similar in each case, so we can probably track it using this single topic for now :+1:</p>

**Comment by alefragnani** [comment]
<p><a class="mention" href="/lostintangent">@lostintangent</a> no, it’s on an iPad, using a Bluetooth mouse.</p>
<p>I updated my previous comment to be more clear. Sorry about that.</p>
<p>Thank you</p>

**Comment by thekeith** [comment]
<p>Going to +1 on this as well. Does not function on both the magic keyboard trackpad, Magic Trackpad and Bluetooth mouse, on any browser.</p>

**Comment by mikeseeh** [comment]
<p>+1 iOS 14 with Magic Keyboard. Scrolling is not working correctly.</p>

**Comment by ergoamox** [comment]
<p>I have the same problem with Logitech Combo Touch and iPadOS 13.6.1</p>

**Comment by gaijinjoe** [comment]
<p>+1 with the magic keyboard’s trackpad.</p>

**Comment by arjunkomath** [comment]
<p>+1, Unable to scroll the editor while using magic keyboard trackpad</p>

**Comment by matthewisabel** [comment, marked as answer]
<p>Scrolling with the trackpad is fixed with changes in Safari for iOS 14.5. While the ship date of iOS 14.5 is not yet announced, a beta version of iOS 14.5 is available through <a href="https://beta.apple.com/sp/betaprogram/" rel="noopener nofollow ugc">Apple’s beta program</a>. Keep in mind, via the beta program you cannot just selectively update Safari, but your whole OS to 14.5.</p>
<p>I have been running this for a bit and it’s quite nice with the fix (although still a few more things to go, we know) :slight_smile:</p>

**Reply by huw** [reply]
Just FYI, this issue is reoccurring on the VS Code side on iPadOS 16, but seemingly only under specific circumstances (microsoft/vscode#164220). Workarounds with limited success are detailed in the thread for anyone stumbling on this.

**Reply by NoahRR** [reply]
Also experiencing this again. Was able to workaround it with some fiddling for a couple days. But now even after restarting the codespace, safari, and iPad, the scrolling won't work via magic keyboard trackpad OR bluetooth logitech mouse.

**Comment by huw** [comment]
<p>I’m marking this as a solution because it solved the scrolling issue. I noticed you can’t drag to select text, but I’m not bothered by it enough to open a separate thread for now.</p>
<p>For those on iPadOS 14.5 Beta and trying to use the feature, it only works when the URL bar is hidden. You’ll either need to scroll a little bit if you’re in Safari, or add your Codespace to the home screen (definitely preferred since you won’t run into any keyboard shortcut conflicts).</p>

**Comment by clouderops** [comment]
I am on ios 16.3 and scrolling with magic keyboard on a ipad does not seem to work.

**Reply by huw** [reply]
Check out the [linked bug](https://github.com/microsoft/vscode/issues/164220)—is it still broken for you with the ‘Shortcuts’ option enabled in settings and the shortcut bar fully expanded?

**Reply by clouderops** [reply]
Shortcuts option was already enabled for me, turned it off, same issue, turned it back on, no difference.

**Reply by clouderops** [reply]
interesting however, if I expand the keyboards shortcut bar, scrolling works, bit annoying as it auto minimise.

**Reply by huw** [reply]
Yeah, it needs to be expanded on 16.3 now. It gets in the way. Mine doesn’t auto-minimise, but I don’t remember there being a setting for it. Such a weird bug…

**Reply by cornel-sys** [reply]
Really helped, it is very strange, some days ago, it worked with text bar minimised.(I just remembered it, I did update yesterday the iPadOS to 16.3, now I regret it…) Very strange, hope it is going to be fixed in the next updates.

**Reply by jbcaillau** [reply]
hi, same bug on code spaces:
- ipad M1
- ipad os 16.3
- magic keyboard
- scrolling works only with keyboard bar expanded

**Reply by DominikKor** [reply]
+1, I still have this issue on my iPad 7 with Logitech Combo Touch touchpad (two finger scrolling)
